hist/histv7/inc/ROOT/RVariableBinAxis.hxx

```diff
@@ -41,12 +42,21 @@ class RVariableBinAxis final {
 public:
    /// Construct an axis object with variable bins.
    ///
-   /// \param[in] binEdges the (ordered) edges of the normal bins
+   /// \param[in] binEdges the (ordered) edges of the normal bins, must define at least one bin (i.e. size >= 2)
    /// \param[in] enableFlowBins whether to enable underflow and overflow bins
    RVariableBinAxis(std::vector<double> binEdges, bool enableFlowBins = true)
       : fBinEdges(std::move(binEdges)), fEnableFlowBins(enableFlowBins)
    {
-      // FIXME: should validate that fBinEdges is sorted
+      if (fBinEdges.size() < 2) {
+         throw std::invalid_argument("must have >= 2 edges");
```

Comment:
```suggestion
         throw std::invalid_argument("must have >= 2 edges, but has " + std::to_string(fBinEdges.size()));
```

Reply:
For the same reason, I would argue that the user should be able to find the constructor with zero or only one edge.

Reply:
Agreed, although, assuming that the stacktrace is not printed, then the error might be cryptic. Imagine that you have 2000-lines long script and you get:

```
root [0] 
Processing crash.cxx...
terminate called after throwing an instance of 'std::invalid_argument'
  what():  must have 2 edges
```

Where is this message coming from, which class? Maybe saying "bin edges" instead of "edges" is more helpful. At least one then know it's the hist class and not the TRectangle class that complained.

Maybe it's because I'm used to the more verbose message such as
Error in TTree::ParseLeaf leaf XX bla bla...
where one gets detailed info.